packages/dd-trace/src/log_channels.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const dc = require('diagnostics_channel')
+
+const Level = {
+  Debug: 'debug',
+  Info: 'info',
+  Warn: 'warn',
+  Error: 'error'
+}
+
+const _defaultLevel = Level.Debug
+
+class LogChannel {
+  constructor (name, logLevel) {
+    this.channel = dc.channel(`dd-trace:log:${name}`)
+    this.logLevel = logLevel
+  }
+
+  publish (message) {
+    if (this.channel.hasSubscribers) {
+      return this.channel.publish(message)
+    }
+  }
+
+  subscribe (onMessage) {
+    this.channel.subscribe(onMessage)
+  }
+
+  unsubscribe (onMessage) {
+    if (this.channel.hasSubscribers) {
+      this.channel.unsubscribe(onMessage)
+    }
+  }
+}
```

Comment:
This could be simplified to:

```suggestion
class LogChannel extends dc.Channel {
  constructor (name, logLevel) {
    super(`dd-trace:log:${name}`)
    this.logLevel = logLevel
  }
}
```

Reply:
Much better, but wouldn't we be losing the `hasSubscribers` check in `publish` and `unsubscribe` methods?

Reply:
The publish method is a no-op if there's no subscribers so there's no downside to just calling it directly. The `hasSubscribers` helper only exists to allow the producer to skip construction of the input object if they're doing more than just passing a reference through directly, but that's no happening here so it's unnecessary.

Reply:
done